### DISCUSSION TITLE
Getting `setting the PIPES_AS_CONCAT sql_mode is unsupported` error

### DISCUSSION THREAD

**Discussion by marcustut** [planetscale/discussion#248] [Bug Reports] [answered]
I'm using PlanetScale with rust's sqlx. The issue reported can be seen here https://github.com/launchbadge/sqlx/issues/2034

I was able to use sqlx without any issues few days back but when I try to compile the code today, it gave this error. From what I understand, sqlx tries to set `PIPES_AS_CONCAT` as one of the `sql_mode` but planetscale does not allow it.

Below is the output when I'm trying to manually set in through the shell.

```sh
marcustut/development> show variables like '%sql_mode%';
+---------------+-----------------------------------------------------------------------------------------------------------------------+
| Variable_name | Value                                                                                                                 |
+---------------+-----------------------------------------------------------------------------------------------------------------------+
| sql_mode      | ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION |
+---------------+-----------------------------------------------------------------------------------------------------------------------+
marcustut/development> SET sql_mode=(SELECT CONCAT(@@sql_mode, ',PIPES_AS_CONCAT,NO_ENGINE_SUBSTITUTION')),time_zone='+00:00';
ERROR 1235 (42000): setting the PIPES_AS_CONCAT sql_mode is unsupported
```

**Comment by fooforge** [comment, marked as answer]
@marcustut I was able to reproduce this, and, I think, it's got something to do with a Vitess upgrade.

I'm going to open an internal bug report and will be in touch again when I have more to share.

**Reply by marcustut** [reply]
Okay, thanks for your swift response!

**Reply by dbussink** [reply]
I have added an explanation in https://github.com/launchbadge/sqlx/issues/2034#issuecomment-1210594227 of why this is happening. In short, this change is actually a bug fix on our side to avoid silently wrong behavior. See also that issue for more details. 

The fix would be to see if `sqlx` can be changed to avoid using `PIPES_AS_CONCAT`.

**Comment by fooforge** [comment]
@marcustut Thanks for your help with getting this changed to a toggleable option in sqlx (https://github.com/launchbadge/sqlx/pull/2046)! ✨

**Reply by marcustut** [reply]
No problem!
